spec/controllers/shipments_controller_spec.rb

```diff
@@ -0,0 +1,134 @@
+require 'rails_helper'
+require 'pry'
+
+RSpec.describe ShipmentsController, type: :controller do
+  let(:successful_params) do
```

Comment:
You could DRY these let statements up considerably by defining individual parameters and calling them from within other let statements. For example:
<pre>
  let(:destination) do
    { country: "US", city: "Seattle", "state" => "WA", "zip" => 98105 }
  end

  let(:packages) do
    [
      { "weight" => 400, "dimensions" => "4, 5, 6" },
      { "weight" => 100, "dimensions" => "6, 6, 8" }
    ]
  end

  let (:no_origin_params) do
    { destination: destination, packages: packages }
  end
</pre>

By doing it this way, if your requirements for shipping info change, you'll only have to change it in a couple places to update the tests -- instead of having to go through every line to update each reference.